Lagrange.Core/Message/MessageFilter.cs

```diff
@@ -10,33 +10,60 @@ internal static class MessageFilter
     /// <summary>
     /// The filter rules, result of the predicate is the index of the message entity that should be removed, -1 means the message entity should be kept.
     /// </summary>
-    private static readonly List<Func<MessageChain, int>> FilterRules = new();
+    private static readonly List<Func<MessageChain, (int index, bool isCompleted)>> FilterRules = new();
 
     static MessageFilter()
-    { 
+    {
+        FilterRules.Add(chain =>
+        {
+            var forwardIndex = chain.FindIndex(entity => entity is ForwardEntity);
+
+            if (chain[forwardIndex + 1] is MentionEntity mention) return (forwardIndex + 1, true);
+
+            return (-1, true);
+        });
+
         FilterRules.Add(x =>
         {
-            var forwards = x.OfType<ForwardEntity>().ToArray();
-            var mentions = x.OfType<MentionEntity>().ToArray();
-            
-            foreach (var forward in forwards)
+            var images = x.OfType<ImageEntity>().ToArray();
+
+            for (int i = 0; i < images.Length; i++)
             {
-                foreach (var mention in mentions)
+                var imageOld = images[i];
+                if (!Uri.IsWellFormedUriString(imageOld.ImageUrl, UriKind.RelativeOrAbsolute))
+                {
+                    return (x.IndexOf(imageOld), false);
+                }
+
+                var uri = new Uri(imageOld.ImageUrl);
```

Comment:
This filtering rule **filters out all images** for Emoji sent under the official client TIM `v3.5.8.3228`, I'll provide the Protobuf with the original message that can reproduce this